diff -ruN cm/front/AndroidProducts.mk mokee/front/AndroidProducts.mk
--- cm/front/AndroidProducts.mk	2013-07-05 16:19:19.766287748 +0800
+++ mokee/front/AndroidProducts.mk	2013-07-05 16:20:10.465083863 +0800
@@ -1,2 +1,2 @@
 PRODUCT_MAKEFILES := \
-    $(LOCAL_DIR)/device_front.mk
+    $(LOCAL_DIR)/full_front.mk
diff -ruN cm/front/bltsville/gcbv/Android.mk mokee/front/bltsville/gcbv/Android.mk
--- cm/front/bltsville/gcbv/Android.mk	2013-07-05 16:19:19.729288628 +0800
+++ mokee/front/bltsville/gcbv/Android.mk	2013-07-05 16:20:10.443084384 +0800
@@ -45,10 +45,10 @@
 LOCAL_C_INCLUDES := \
 	$(LOCAL_PATH)/mirror \
 	$(LOCAL_PATH)/mirror/include \
-	$(LOCAL_PATH)/../bltsville/include \
-	$(LOCAL_PATH)/../ocd/include
+	$(DEVICE_FOLDER)/bltsville/bltsville/include \
+	$(DEVICE_FOLDER)/bltsville/ocd/include
 
-VERSION_H := $(LOCAL_PATH)/version.h
+VERSION_H := $(DEVICE_FOLDER)/bltsville/gcbv/version.h
 BV_VERSION := $(shell grep "VER_FILEVERSION_STR" $(VERSION_H) | sed "s,.*\"\([0-9.]*\)\\\0.*,\1,")
 
 LOCAL_SHARED_LIBRARIES := \
diff -ruN cm/front/bltsville/ticpu/Android.mk mokee/front/bltsville/ticpu/Android.mk
--- cm/front/bltsville/ticpu/Android.mk	2013-07-05 16:19:19.716288938 +0800
+++ mokee/front/bltsville/ticpu/Android.mk	2013-07-05 16:20:10.415085048 +0800
@@ -18,7 +18,7 @@
 
 #Copying libbltsville_ticpu.BV_CPUVERSION.so
 include $(CLEAR_VARS)
-#BV_CPUVERSION :=$(shell ls $(COMMON_PATH)/bltsville/ticpu/lib/android/libbltsville_*.*.so|\
+#BV_CPUVERSION :=$(shell ls $(DEVICE_FOLDER)/bltsville/ticpu/lib/android/libbltsville_*.*.so|\
 #             sed 's/device\/samsung\/omap4-common\/bltsville\/ticpu\/lib\/android\/libbltsville_ticpu.//')
 BV_CPUVERSION = "2.1.0.0.so"
 LOCAL_MODULE_TAGS := optional
@@ -38,7 +38,7 @@
 	@mkdir -p $(dir $@)
 	@rm -rf $@
 	$(hide) ln -fs $(TICPU_BINARY) $@
-	@cp -afr $(COMMON_PATH)/bltsville/ticpu/lib/android/libbltsville_ticpu_license.txt $(TARGET_OUT_VENDOR)/lib
+	@cp -afr $(DEVICE_FOLDER)/bltsville/ticpu/lib/android/libbltsville_ticpu_license.txt $(TARGET_OUT_VENDOR)/lib
 ALL_DEFAULT_INSTALLED_MODULES += $(SYMLINKS)
 
 SYMLINKS1 := $(TARGET_OUT_VENDOR)/lib/libbltsville_cpu.so
diff -ruN cm/front/cm.dependencies mokee/front/cm.dependencies
--- cm/front/cm.dependencies	2013-07-05 16:19:19.754288033 +0800
+++ mokee/front/cm.dependencies	1970-01-01 08:00:00.000000000 +0800
@@ -1,7 +0,0 @@
-[
- {
-   "repository": "android_kernel_huawei_front",
-   "target_path": "kernel/huawei/front",
-   "branch": "jellybean"
- }
-]
diff -ruN cm/front/cm.mk mokee/front/cm.mk
--- cm/front/cm.mk	2013-07-05 16:19:19.814286606 +0800
+++ mokee/front/cm.mk	1970-01-01 08:00:00.000000000 +0800
@@ -1,27 +0,0 @@
-# Inherit some common CM stuff.
-$(call inherit-product, vendor/cm/config/common_full_phone.mk)
-$(call inherit-product, vendor/cm/config/gsm.mk)
-
-$(call inherit-product, device/huawei/front/device_front.mk)
-
-# Release name
-PRODUCT_RELEASE_NAME := front
-
-# Preload bootanimation
-TARGET_BOOTANIMATION_PRELOAD := true
-TARGET_BOOTANIMATION_TEXTURE_CACHE := true
-
-# Boot animation
-TARGET_SCREEN_HEIGHT := 1280
-TARGET_SCREEN_WIDTH := 720
-
-#Set build fingerprint / ID / Product Name ect.
-#PRODUCT_BUILD_PROP_OVERRIDES += PRODUCT_NAME=front BUILD_FINGERPRINT="Huawei/front/front:4.2.1:user/release-keys" PRIVATE_BUILD_DESC="U9500-1V100R001RUSC00B522_SYSTEM"
-
-## Device identifier. This must come after all inclusions
-PRODUCT_DEVICE := front
-PRODUCT_NAME := cm_front
-PRODUCT_RELEASE_NAME := U9500
-PRODUCT_BRAND := Huawei
-PRODUCT_MODEL := U9500
-PRODUCT_MANUFACTURER := Huawei
diff -ruN cm/front/device_front.mk mokee/front/device_front.mk
--- cm/front/device_front.mk	2013-07-05 16:19:19.869285297 +0800
+++ mokee/front/device_front.mk	1970-01-01 08:00:00.000000000 +0800
@@ -1,183 +0,0 @@
-#
-# Copyright (C) 2012 The Android Open-Source Project
-# Copyright (C) 2012 mdeejay
-# Copyright (C) 2013 faust93
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-$(call inherit-product, $(SRC_TARGET_DIR)/product/languages_full.mk)
-$(call inherit-product-if-exists, vendor/huawei/front/front-vendor.mk)
-$(call inherit-product, hardware/ti/omap4xxx/omap4.mk)
-$(call inherit-product, $(SRC_TARGET_DIR)/product/full_base_telephony.mk)
-$(call inherit-product, frameworks/native/build/phone-xhdpi-1024-dalvik-heap.mk)
-
-
-# This file includes all definitions that apply to ALL front devices, and
-# are also specific to front devices
-#
-# Everything in this directory will become public
-
-DEVICE_PACKAGE_OVERLAYS := device/huawei/front/overlay
-
-#Camera
-PRODUCT_PACKAGES += \
-        Camera
-
-# Audio Packages
-PRODUCT_PACKAGES += \
-	audio.a2dp.default \
-	audio.usb.default \
-	libtinyalsa \
-	tinycap \
-	tinymix \
-	tinyplay \
-	libaudioutils
-
-# PowerHAL
-PRODUCT_PACKAGES += \
-	audio.primary.front \
-	power.front \
-	hwcomposer.front \
-	libion_ti
-
-# Ramdisk
-PRODUCT_COPY_FILES += \
-	device/huawei/front/ramdisk/init:root/init \
-	device/huawei/front/ramdisk/init.rc:root/init.rc \
-	device/huawei/front/ramdisk/init.front.rc:root/init.front.rc \
-	device/huawei/front/ramdisk/init.front.usb.rc:root/init.front.usb.rc \
-	device/huawei/front/ramdisk/ueventd.rc:root/ueventd.rc \
-	device/huawei/front/ramdisk/ueventd.front.rc:root/ueventd.front.rc \
-	device/huawei/front/ramdisk/default.prop:root/default.prop
-
-# Media / Audio
-PRODUCT_COPY_FILES += \
-	device/huawei/front/media/media_profiles.xml:system/etc/media_profiles.xml \
-	device/huawei/front/media/media_codecs.xml:system/etc/media_codecs.xml \
-	device/huawei/front/audio/audio_policy.conf:system/etc/audio_policy.conf \
-	device/huawei/front/audio/huawei/audio/cm_front_audio_config.conf:system/etc/huawei/audio/cm_front_audio_config.conf \
-	device/huawei/front/audio/huawei/audio/front_audio_config.conf:system/etc/huawei/audio/front_audio_config.conf \
-	device/huawei/front/audio/huawei/audio/front_factory_audio_config.conf:system/etc/huawei/audio/front_factory_audio_config.conf \
-	device/huawei/front/audio/huawei/audio/U9500_audio_config.conf:system/etc/huawei/audio/U9500_audio_config.conf \
-	device/huawei/front/audio/huawei/audio/U9500_factory_audio_config.conf:system/etc/huawei/audio/U9500_factory_audio_config.conf
-
-# GPS
-PRODUCT_COPY_FILES += \
-	device/huawei/front/gps/gps.conf:system/etc/gps.conf \
-	device/huawei/front/gps/gpsconfig.xml:system/etc/gpsconfig.xml
-
-# BT
-PRODUCT_COPY_FILES += \
-	device/huawei/front/bluetooth/audio.conf:system/etc/bluetooth/audio.conf \
-	device/huawei/front/bluetooth/auto_pairing.conf:system/etc/bluetooth/auto_pairing.conf \
-	device/huawei/front/bluetooth/BCM4330B1.hcd:system/etc/bluetooth/BCM4330B1.hcd \
-	device/huawei/front/bluetooth/blacklist.conf:system/etc/bluetooth/blacklist.conf \
-	device/huawei/front/bluetooth/init.bcm.chip_off.sh:system/etc/bluetooth/init.bcm.chip_off.sh \
-	device/huawei/front/bluetooth/init.bcm.chip_on.sh:system/etc/bluetooth/init.bcm.chip_on.sh \
-	device/huawei/front/bluetooth/input.conf:system/etc/bluetooth/input.conf \
-	device/huawei/front/bluetooth/network.conf:system/etc/bluetooth/network.conf
-
-# Vold
-PRODUCT_COPY_FILES += \
-	device/huawei/front/vold.fstab:system/etc/vold.fstab
-
-# Torch
-PRODUCT_PACKAGES += \
-        Torch
-
-# Live Wallpapers
-PRODUCT_PACKAGES += \
-        LiveWallpapers \
-        LiveWallpapersPicker \
-        VisualizationWallpapers \
-        librs_jni
-
-# Common
-PRODUCT_PACKAGES += \
-	make_ext4fs \
-	setup_fs \
-	l2ping \
-	com.android.future.usb.accessory
-
-# Key maps
-PRODUCT_COPY_FILES += \
-	device/huawei/front/usr/omap4-keypad.kl:system/usr/keylayout/omap4-keypad.kl \
-	device/huawei/front/usr/omap4-keypad.kcm:system/usr/keychars/omap4-keypad.kcm \
-	device/huawei/front/usr/twl6030_pwrbutton.kl:system/usr/keylayout/twl6030_pwrbutton.kl
-
-# Input device calibration files
-PRODUCT_COPY_FILES += \
-	device/huawei/front/usr/syn_tm12xx_ts_1.idc:system/usr/idc/syn_tm12xx_ts_1.idc \
-	device/huawei/front/usr/syn_tm12xx_ts_2.idc:system/usr/idc/syn_tm12xx_ts_2.idc
-
-# These are the hardware-specific features
-PRODUCT_COPY_FILES += \
-	frameworks/native/data/etc/handheld_core_hardware.xml:system/etc/permissions/handheld_core_hardware.xml \
-	frameworks/native/data/etc/android.hardware.telephony.gsm.xml:system/etc/permissions/android.hardware.telephony.gsm.xml \
-	frameworks/native/data/etc/android.hardware.bluetooth.xml:system/etc/permissions/android.hardware.bluetooth.xml \
-	frameworks/native/data/etc/android.hardware.camera.flash-autofocus.xml:system/etc/permissions/android.hardware.camera.flash-autofocus.xml \
-	frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
-	frameworks/native/data/etc/android.hardware.location.gps.xml:system/etc/permissions/android.hardware.location.gps.xml \
-	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
-	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
-	frameworks/native/data/etc/android.hardware.sensor.accelerometer.xml:system/etc/permissions/android.hardware.sensor.accelerometer.xml \
-	frameworks/native/data/etc/android.hardware.sensor.proximity.xml:system/etc/permissions/android.hardware.sensor.proximity.xml \
-	frameworks/native/data/etc/android.hardware.sensor.light.xml:system/etc/permissions/android.hardware.sensor.light.xml \
-	frameworks/native/data/etc/android.hardware.sensor.gyroscope.xml:system/etc/permissions/android.hardware.sensor.gyroscope.xml \
-	frameworks/native/data/etc/android.hardware.sensor.barometer.xml:system/etc/permissions/android.hardware.sensor.barometer.xml \
-	frameworks/native/data/etc/android.hardware.touchscreen.multitouch.jazzhand.xml:system/etc/permissions/android.hardware.touchscreen.multitouch.jazzhand.xml \
-	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
-	frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml \
-	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml
-
-PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
-packages/wallpapers/LivePicker/android.software.live_wallpaper.xml:system/etc/permissions/android.software.live_wallpaper.xml)
-
-PRODUCT_PACKAGES += \
-	librs_jni \
-	com.android.future.usb.accessory
-
-# Filesystem management tools
-PRODUCT_PACKAGES += \
-	make_ext4fs \
-	e2fsck \
-	setup_fs
-
-#Lib Skia test
-PRODUCT_PACKAGES += \
-	SkLibTiJpeg_Test
-
-# for bugmailer
-PRODUCT_PACKAGES += send_bug
-PRODUCT_COPY_FILES += \
-	system/extras/bugmailer/bugmailer.sh:system/bin/bugmailer.sh \
-	system/extras/bugmailer/send_bug:system/bin/send_bug
-
-PRODUCT_PROPERTY_OVERRIDES += \
-	ro.setupwizard.enable_bypass=1 \
-	dalvik.vm.dexopt-flags=m=y \
-	ro.product.board=U9500
-
-PRODUCT_DEFAULT_PROPERTY_OVERRIDES += \
-	ro.secure=0
-
-PRODUCT_CHARACTERISTICS := default
-#PRODUCT_BUILD_PROP_OVERRIDES += BUILD_UTC_DATE=0
-#PRODUCT_TAGS += dalvik.gc.type-precise
-PRODUCT_AAPT_CONFIG := normal hdpi
-PRODUCT_AAPT_PREF_CONFIG := hdpi
-PRODUCT_LOCALES += en_US hdpi
-BOARD_WLAN_DEVICE_REV := bcm4330_b1
-WIFI_BAND             := 802_11_ABG
-
diff -ruN cm/front/edid/Android.mk mokee/front/edid/Android.mk
--- cm/front/edid/Android.mk	2013-07-05 16:19:19.785287296 +0800
+++ mokee/front/edid/Android.mk	2013-07-05 16:20:10.508082844 +0800
@@ -12,8 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# AOSP specific
-ifneq ($(TARGET_PRODUCT),full_$(TARGET_BOOTLOADER_BOARD_NAME))
+ifeq ($(TARGET_DEVICE),front)
 
 LOCAL_PATH:= $(call my-dir)
 
diff -ruN cm/front/egl.cfg mokee/front/egl.cfg
--- cm/front/egl.cfg	2013-07-05 16:19:19.870285273 +0800
+++ mokee/front/egl.cfg	2013-07-05 16:20:10.597080735 +0800
@@ -1,2 +1 @@
-0 0 android
 0 1 POWERVR_SGX540_120
\ No newline at end of file
diff -ruN cm/front/full_front.mk mokee/front/full_front.mk
--- cm/front/full_front.mk	1970-01-01 08:00:00.000000000 +0800
+++ mokee/front/full_front.mk	2013-07-05 18:25:02.523727066 +0800
@@ -0,0 +1,184 @@
+#
+# Copyright (C) 2012 The Android Open-Source Project
+# Copyright (C) 2012 mdeejay
+# Copyright (C) 2013 faust93
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+$(call inherit-product, $(SRC_TARGET_DIR)/product/languages_full.mk)
+$(call inherit-product-if-exists, vendor/huawei/front/front-vendor.mk)
+$(call inherit-product, hardware/ti/omap4xxx/omap4.mk)
+$(call inherit-product, $(SRC_TARGET_DIR)/product/full_base_telephony.mk)
+$(call inherit-product, frameworks/native/build/phone-xhdpi-1024-dalvik-heap.mk)
+
+DEVICE_FOLDER := device/huawei/front
+
+# This file includes all definitions that apply to ALL front devices, and
+# are also specific to front devices
+#
+# Everything in this directory will become public
+
+DEVICE_PACKAGE_OVERLAYS := device/huawei/front/overlay
+
+#Camera
+PRODUCT_PACKAGES += \
+        Camera
+
+# Audio Packages
+PRODUCT_PACKAGES += \
+	audio.a2dp.default \
+	audio.usb.default \
+	libtinyalsa \
+	tinycap \
+	tinymix \
+	tinyplay \
+	libaudioutils
+
+# PowerHAL
+PRODUCT_PACKAGES += \
+	audio.primary.front \
+	power.front \
+	hwcomposer.front \
+	libion_ti
+
+# Ramdisk
+PRODUCT_COPY_FILES += \
+	device/huawei/front/ramdisk/init:root/init \
+	device/huawei/front/ramdisk/init.rc:root/init.rc \
+	device/huawei/front/ramdisk/init.front.rc:root/init.front.rc \
+	device/huawei/front/ramdisk/init.front.usb.rc:root/init.front.usb.rc \
+	device/huawei/front/ramdisk/ueventd.rc:root/ueventd.rc \
+	device/huawei/front/ramdisk/ueventd.front.rc:root/ueventd.front.rc \
+	device/huawei/front/ramdisk/default.prop:root/default.prop
+
+# Media / Audio
+PRODUCT_COPY_FILES += \
+	device/huawei/front/media/media_profiles.xml:system/etc/media_profiles.xml \
+	device/huawei/front/media/media_codecs.xml:system/etc/media_codecs.xml \
+	device/huawei/front/audio/audio_policy.conf:system/etc/audio_policy.conf \
+	device/huawei/front/audio/huawei/audio/cm_front_audio_config.conf:system/etc/huawei/audio/mk_front_audio_config.conf \
+	device/huawei/front/audio/huawei/audio/front_audio_config.conf:system/etc/huawei/audio/front_audio_config.conf \
+	device/huawei/front/audio/huawei/audio/front_factory_audio_config.conf:system/etc/huawei/audio/front_factory_audio_config.conf \
+	device/huawei/front/audio/huawei/audio/U9500_audio_config.conf:system/etc/huawei/audio/U9500_audio_config.conf \
+	device/huawei/front/audio/huawei/audio/U9500_factory_audio_config.conf:system/etc/huawei/audio/U9500_factory_audio_config.conf
+
+# GPS
+PRODUCT_COPY_FILES += \
+	device/huawei/front/gps/gps.conf:system/etc/gps.conf \
+	device/huawei/front/gps/gpsconfig.xml:system/etc/gpsconfig.xml
+
+# BT
+PRODUCT_COPY_FILES += \
+	device/huawei/front/bluetooth/audio.conf:system/etc/bluetooth/audio.conf \
+	device/huawei/front/bluetooth/auto_pairing.conf:system/etc/bluetooth/auto_pairing.conf \
+	device/huawei/front/bluetooth/BCM4330B1.hcd:system/etc/bluetooth/BCM4330B1.hcd \
+	device/huawei/front/bluetooth/blacklist.conf:system/etc/bluetooth/blacklist.conf \
+	device/huawei/front/bluetooth/init.bcm.chip_off.sh:system/etc/bluetooth/init.bcm.chip_off.sh \
+	device/huawei/front/bluetooth/init.bcm.chip_on.sh:system/etc/bluetooth/init.bcm.chip_on.sh \
+	device/huawei/front/bluetooth/input.conf:system/etc/bluetooth/input.conf \
+	device/huawei/front/bluetooth/network.conf:system/etc/bluetooth/network.conf
+
+# Vold
+PRODUCT_COPY_FILES += \
+	device/huawei/front/vold.fstab:system/etc/vold.fstab
+
+# Torch
+PRODUCT_PACKAGES += \
+        Torch
+
+# Live Wallpapers
+PRODUCT_PACKAGES += \
+        LiveWallpapers \
+        LiveWallpapersPicker \
+        VisualizationWallpapers \
+        librs_jni
+
+# Common
+PRODUCT_PACKAGES += \
+	make_ext4fs \
+	setup_fs \
+	l2ping \
+	com.android.future.usb.accessory
+
+# Key maps
+PRODUCT_COPY_FILES += \
+	device/huawei/front/usr/omap4-keypad.kl:system/usr/keylayout/omap4-keypad.kl \
+	device/huawei/front/usr/omap4-keypad.kcm:system/usr/keychars/omap4-keypad.kcm \
+	device/huawei/front/usr/twl6030_pwrbutton.kl:system/usr/keylayout/twl6030_pwrbutton.kl
+
+# Input device calibration files
+PRODUCT_COPY_FILES += \
+	device/huawei/front/usr/syn_tm12xx_ts_1.idc:system/usr/idc/syn_tm12xx_ts_1.idc \
+	device/huawei/front/usr/syn_tm12xx_ts_2.idc:system/usr/idc/syn_tm12xx_ts_2.idc
+
+# These are the hardware-specific features
+PRODUCT_COPY_FILES += \
+	frameworks/native/data/etc/handheld_core_hardware.xml:system/etc/permissions/handheld_core_hardware.xml \
+	frameworks/native/data/etc/android.hardware.telephony.gsm.xml:system/etc/permissions/android.hardware.telephony.gsm.xml \
+	frameworks/native/data/etc/android.hardware.bluetooth.xml:system/etc/permissions/android.hardware.bluetooth.xml \
+	frameworks/native/data/etc/android.hardware.camera.flash-autofocus.xml:system/etc/permissions/android.hardware.camera.flash-autofocus.xml \
+	frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
+	frameworks/native/data/etc/android.hardware.location.gps.xml:system/etc/permissions/android.hardware.location.gps.xml \
+	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
+	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
+	frameworks/native/data/etc/android.hardware.sensor.accelerometer.xml:system/etc/permissions/android.hardware.sensor.accelerometer.xml \
+	frameworks/native/data/etc/android.hardware.sensor.proximity.xml:system/etc/permissions/android.hardware.sensor.proximity.xml \
+	frameworks/native/data/etc/android.hardware.sensor.light.xml:system/etc/permissions/android.hardware.sensor.light.xml \
+	frameworks/native/data/etc/android.hardware.sensor.gyroscope.xml:system/etc/permissions/android.hardware.sensor.gyroscope.xml \
+	frameworks/native/data/etc/android.hardware.sensor.barometer.xml:system/etc/permissions/android.hardware.sensor.barometer.xml \
+	frameworks/native/data/etc/android.hardware.touchscreen.multitouch.jazzhand.xml:system/etc/permissions/android.hardware.touchscreen.multitouch.jazzhand.xml \
+	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
+	frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml \
+	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml
+
+PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
+packages/wallpapers/LivePicker/android.software.live_wallpaper.xml:system/etc/permissions/android.software.live_wallpaper.xml)
+
+PRODUCT_PACKAGES += \
+	librs_jni \
+	com.android.future.usb.accessory
+
+# Filesystem management tools
+PRODUCT_PACKAGES += \
+	make_ext4fs \
+	e2fsck \
+	setup_fs
+
+#Lib Skia test
+PRODUCT_PACKAGES += \
+	SkLibTiJpeg_Test
+
+# for bugmailer
+PRODUCT_PACKAGES += send_bug
+PRODUCT_COPY_FILES += \
+	system/extras/bugmailer/bugmailer.sh:system/bin/bugmailer.sh \
+	system/extras/bugmailer/send_bug:system/bin/send_bug
+
+PRODUCT_PROPERTY_OVERRIDES += \
+	ro.setupwizard.enable_bypass=1 \
+	dalvik.vm.dexopt-flags=m=y \
+	ro.product.board=U9500
+
+PRODUCT_DEFAULT_PROPERTY_OVERRIDES += \
+	ro.secure=0
+
+PRODUCT_CHARACTERISTICS := default
+#PRODUCT_BUILD_PROP_OVERRIDES += BUILD_UTC_DATE=0
+#PRODUCT_TAGS += dalvik.gc.type-precise
+PRODUCT_AAPT_CONFIG := normal hdpi
+PRODUCT_AAPT_PREF_CONFIG := hdpi
+PRODUCT_LOCALES += en_US hdpi
+BOARD_WLAN_DEVICE_REV := bcm4330_b1
+WIFI_BAND             := 802_11_ABG
+
diff -ruN cm/front/mk.dependencies mokee/front/mk.dependencies
--- cm/front/mk.dependencies	1970-01-01 08:00:00.000000000 +0800
+++ mokee/front/mk.dependencies	2013-07-05 16:20:10.481083483 +0800
@@ -0,0 +1,7 @@
+[
+ {
+   "repository": "android_kernel_huawei_front",
+   "target_path": "kernel/huawei/front",
+   "branch": "jellybean"
+ }
+]
diff -ruN cm/front/mokee.mk mokee/front/mokee.mk
--- cm/front/mokee.mk	1970-01-01 08:00:00.000000000 +0800
+++ mokee/front/mokee.mk	2013-07-05 16:20:10.581081114 +0800
@@ -0,0 +1,27 @@
+# Inherit some common MK stuff.
+$(call inherit-product, vendor/mk/config/common_full_phone.mk)
+$(call inherit-product, vendor/mk/config/gsm.mk)
+
+$(call inherit-product, device/huawei/front/full_front.mk)
+
+# Release name
+PRODUCT_RELEASE_NAME := front
+
+# Preload bootanimation
+TARGET_BOOTANIMATION_PRELOAD := true
+TARGET_BOOTANIMATION_TEXTURE_CACHE := true
+
+# Boot animation
+TARGET_SCREEN_HEIGHT := 1280
+TARGET_SCREEN_WIDTH := 720
+
+#Set build fingerprint / ID / Product Name ect.
+#PRODUCT_BUILD_PROP_OVERRIDES += PRODUCT_NAME=front BUILD_FINGERPRINT="Huawei/front/front:4.2.1:user/release-keys" PRIVATE_BUILD_DESC="U9500-1V100R001RUSC00B523_SYSTEM"
+
+## Device identifier. This must come after all inclusions
+PRODUCT_DEVICE := front
+PRODUCT_NAME := mk_front
+PRODUCT_RELEASE_NAME := U9500
+PRODUCT_BRAND := Huawei
+PRODUCT_MODEL := U9500
+PRODUCT_MANUFACTURER := Huawei
diff -ruN cm/front/ramdisk/default.prop mokee/front/ramdisk/default.prop
--- cm/front/ramdisk/default.prop	2013-07-05 16:19:19.863285440 +0800
+++ mokee/front/ramdisk/default.prop	2013-07-05 16:20:10.596080758 +0800
@@ -2,8 +2,8 @@
 # ADDITIONAL_DEFAULT_PROPERTIES
 #
 ro.secure=0
-ro.allow.mock.location=1
+ro.allow.mock.location=0
 ro.debuggable=1
 ro.adb.secure=0
 persist.sys.usb.config=mtp,mass_storage,adb
-ro.build.display.id=ppyyf_build
+ro.build.display.id=ppyyf_mk_build
diff -ruN cm/front/vendorsetup.sh mokee/front/vendorsetup.sh
--- cm/front/vendorsetup.sh	2013-07-05 16:19:19.771287629 +0800
+++ mokee/front/vendorsetup.sh	2013-07-05 16:20:10.468083791 +0800
@@ -20,5 +20,5 @@
 # In particular, you can add lunch options with the add_lunch_combo
 # function: add_lunch_combo generic-eng
 
-add_lunch_combo pa_front-userdebug
-add_lunch_combo pa_front-eng
+#add_lunch_combo pa_front-userdebug
+#add_lunch_combo pa_front-eng
